botocore/client.py

```diff
@@ -174,8 +171,46 @@ def waiter_names(self):
     def _load_service_model(self, service_name):
         json_model = self._loader.load_service_model('aws/%s' % service_name)
         service_model = ServiceModel(json_model, service_name=service_name)
+        self._register_retries(service_model)
         return service_model
 
+    def _register_retries(self, service_model):
+        endpoint_prefix = service_model.endpoint_prefix
+
+        # First, we load the entire retry config for all services,
+        # then pull out just the information we need.
+        original_config = self._loader.load_data('aws/_retry')
+        if not original_config:
+            return
+
+        retry_config = translate.build_retry_config(
+            endpoint_prefix, original_config.get('retry', {}),
+            original_config.get('definitions', {}))
+
+        logger.debug("Registering retry handlers for service: %s",
+                     service_model.service_name)
+        handler = retryhandler.create_retry_handler(
```

Comment:
This is going to require a patch to test.  The client code has been pretty consistent with requiring dependencies being passed in.  It makes customization cleaner and tests easier to read.  I'd like to stick to that convention.  I realize some of the recent signature changes don't follow this, but I'd like to not veer off too far in that direction.